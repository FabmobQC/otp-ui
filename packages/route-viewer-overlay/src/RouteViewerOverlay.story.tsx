import React, { useEffect, useState } from "react";

import routeData from "../__mocks__/mock-route.json";
import routeData2 from "../__mocks__/mock-route2.json";
import routeDataOtp2 from "../__mocks__/mock-route-otp2.json";
import flexRouteData from "../__mocks__/mock-flex-route.json";
import flexRouteData2 from "../__mocks__/mock-flex-route2.json";
import flexRouteData3 from "../__mocks__/mock-flex-route3.json";

import StopsOverlay from "../../stops-overlay/src";
import { withMap } from "../../../.storybook/base-map-wrapper";
import RouteViewerOverlay from ".";

const PORTLAND = [45.543092, -122.671202];
const POWDER_SPRINGS = [33.8595, -84.67483];
const zoom = 11;
const disableStoryshots = { storyshots: { disable: true } };

// Hide some story args completely.
const hiddenProp = {
  table: { disable: true }
};

export default {
  argTypes: {
    extraLayer: hiddenProp,
    path: hiddenProp,
    routeData: hiddenProp
  },
  component: RouteViewerOverlay,
  decorators: [withMap(PORTLAND, zoom)],
  parameters: disableStoryshots,
  title: "RouteViewerOverlay"
};

const Template = args => (
  <>
    <RouteViewerOverlay
      clipToPatternStops={args.clipToPatternStops}
      path={args.path}
      routeData={args.routeData}
    />
    {args.extraLayer}
  </>
);

/** Should zoom to pattern. */

const WithChangingRoute = () => {
  const [rtData, setRouteData] = useState(routeData);

  useEffect(() => {
    const interval = setInterval(() => {
      setRouteData(rtData === routeData2 ? routeData : routeData2);
    }, 5000);

    return () => {
      clearInterval(interval);
    };
  }, [rtData]);

  return <RouteViewerOverlay routeData={rtData} />;
};

export const Default = Template.bind({});
Default.args = {
  routeData
};
Default.parameters = disableStoryshots;

export const WithPathStyling = Template.bind({});
WithPathStyling.args = {
  path: {
    opacity: 0.5,
    weight: 10
  },
  routeData
};
WithPathStyling.parameters = disableStoryshots;

export const OTP2RouteOutsideOfInitialView = Template.bind({});
OTP2RouteOutsideOfInitialView.args = {
  routeData: routeDataOtp2
};
OTP2RouteOutsideOfInitialView.parameters = disableStoryshots;

export const WithChangingPath = (): JSX.Element => <WithChangingRoute />;
WithChangingPath.parameters = disableStoryshots;

export const FlexRoute = Template.bind({});
FlexRoute.args = {
  clipToPatternStops: true,
  // Since the data is fixed, we know where the relevant flex zone is.
  // Using the stopsOverlay is not possible as it is very complex to implement */}
  extraLayer: (
    <StopsOverlay
      stops={flexRouteData.patterns[0].stops
        .filter(stop => stop?.geometries?.geoJson?.type === "Polygon")
        .map(stop => ({ ...stop, color: `#${flexRouteData.color}` }))}
      symbols={[]}
      visible
    />
  ),
  routeData: flexRouteData
};

FlexRoute.argTypes = {
  clipToPatternStops: { control: "boolean" }
};
FlexRoute.decorators = [withMap(POWDER_SPRINGS, zoom)];
FlexRoute.parameters = disableStoryshots;

export const FlexRoute2 = Template.bind({});
FlexRoute2.args = {
  clipToPatternStops: true,
  // Since the data is fixed, we know where the relevant flex zone is.
  // Using the stopsOverlay is not possible as it is very complex to implement */}
  extraLayer: (
    <StopsOverlay
      stops={Object.values(flexRouteData2.patterns)[0]
        .stops.filter(stop => stop?.geometries?.geoJson?.type === "Polygon")
        .map(stop => ({ ...stop, color: `#${flexRouteData2.color}` }))}
      symbols={[]}
      visible
    />
  ),
  routeData: flexRouteData2
};

FlexRoute2.argTypes = {
  clipToPatternStops: { control: "boolean" }
};
FlexRoute2.decorators = [withMap(POWDER_SPRINGS, zoom)];
FlexRoute2.parameters = disableStoryshots;

export const FlexRoute3 = Template.bind({});
FlexRoute3.args = {
  extraLayer: (
    <StopsOverlay
      // Tests stop area with GeometryCollection
      stops={Object.values(flexRouteData3.patterns)[2]
        .stops.filter(
          stop =>
            stop?.geometries?.geoJson?.type === "Polygon" ||
            stop?.geometries?.geoJson?.type === "GeometryCollection"
        )
        .map(stop => ({ ...stop, color: `#333` }))}
      symbols={[]}
      visible
    />
  ),
  routeData: flexRouteData3
};
FlexRoute3.parameters = disableStoryshots;
