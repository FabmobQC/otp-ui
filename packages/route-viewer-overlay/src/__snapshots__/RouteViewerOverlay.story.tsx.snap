// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteViewerOverlay Default smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteViewerOverlay FlexRoute smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
    <div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
      <div class="maplibregl-map mapboxgl-map"
           style="position: relative; width: 100%; height: 100%;"
      >
        <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
          <canvas class="maplibregl-canvas mapboxgl-canvas"
                  tabindex="0"
                  aria-label="Map"
                  role="region"
                  width="1248"
                  height="648"
                  style="width: 1248px; height: 648px;"
          >
          </canvas>
        </div>
        <div class="maplibregl-control-container mapboxgl-control-container">
          <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
          </div>
          <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
          </div>
          <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
          </div>
          <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
            <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                     open
            >
              <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                       title="Toggle attribution"
                       aria-label="Toggle attribution"
              >
              </summary>
              <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
                ©
                <a href="https://carto.com/about-carto/"
                   target="_blank"
                   rel="noopener"
                >
                  CARTO
                </a>
                , ©
                <a href="http://www.openstreetmap.org/about/"
                   target="_blank"
                >
                  OpenStreetMap
                </a>
                contributors
              </div>
            </details>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteViewerOverlay FlexRoute2 smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
    <div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
      <div class="maplibregl-map mapboxgl-map"
           style="position: relative; width: 100%; height: 100%;"
      >
        <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
          <canvas class="maplibregl-canvas mapboxgl-canvas"
                  tabindex="0"
                  aria-label="Map"
                  role="region"
                  width="1248"
                  height="648"
                  style="width: 1248px; height: 648px;"
          >
          </canvas>
        </div>
        <div class="maplibregl-control-container mapboxgl-control-container">
          <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
          </div>
          <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
          </div>
          <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
          </div>
          <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
            <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                     open
            >
              <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                       title="Toggle attribution"
                       aria-label="Toggle attribution"
              >
              </summary>
              <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
                ©
                <a href="https://carto.com/about-carto/"
                   target="_blank"
                   rel="noopener"
                >
                  CARTO
                </a>
                , ©
                <a href="http://www.openstreetmap.org/about/"
                   target="_blank"
                >
                  OpenStreetMap
                </a>
                contributors
              </div>
            </details>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteViewerOverlay FlexRoute3 smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteViewerOverlay OTP2RouteOutsideOfInitialView smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteViewerOverlay WithChangingPath smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteViewerOverlay WithPathStyling smoke-test 1`] = `
<div class="styled__StoryMapContainer-sc-12kjso7-6 jsxXhE">
  <div class="maplibregl-map mapboxgl-map"
       style="position: relative; width: 100%; height: 100%;"
  >
    <div class="maplibregl-canvas-container mapboxgl-canvas-container maplibregl-interactive mapboxgl-interactive maplibregl-touch-drag-pan mapboxgl-touch-drag-pan maplibregl-touch-zoom-rotate mapboxgl-touch-zoom-rotate">
      <canvas class="maplibregl-canvas mapboxgl-canvas"
              tabindex="0"
              aria-label="Map"
              role="region"
              width="1248"
              height="648"
              style="width: 1248px; height: 648px;"
      >
      </canvas>
    </div>
    <div class="maplibregl-control-container mapboxgl-control-container">
      <div class="maplibregl-ctrl-top-left mapboxgl-ctrl-top-left">
      </div>
      <div class="maplibregl-ctrl-top-right mapboxgl-ctrl-top-right">
      </div>
      <div class="maplibregl-ctrl-bottom-left mapboxgl-ctrl-bottom-left">
      </div>
      <div class="maplibregl-ctrl-bottom-right mapboxgl-ctrl-bottom-right">
        <details class="maplibregl-ctrl maplibregl-ctrl-attrib mapboxgl-ctrl mapboxgl-ctrl-attrib"
                 open
        >
          <summary class="maplibregl-ctrl-attrib-button mapboxgl-ctrl-attrib-button"
                   title="Toggle attribution"
                   aria-label="Toggle attribution"
          >
          </summary>
          <div class="maplibregl-ctrl-attrib-inner mapboxgl-ctrl-attrib-inner">
            ©
            <a href="https://carto.com/about-carto/"
               target="_blank"
               rel="noopener"
            >
              CARTO
            </a>
            , ©
            <a href="http://www.openstreetmap.org/about/"
               target="_blank"
            >
              OpenStreetMap
            </a>
            contributors
          </div>
        </details>
      </div>
    </div>
  </div>
</div>
`;
